### PR TITLE
ICU-21816 Add ICU4C samples check to workflow for BRS "Test C Samples"

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21816
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [X] Tests included, if applicable
- [X] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by sven-oly** [comment]
This should automate the tests of ICU4C samples under source for BRS line 40. This is an automation of the [steps here:](https://unicode-org.github.io/icu/processes/release/tasks/healthy-code.html#linux-unix-build-and-test).

I look forward to your feedback.

File: .github/workflows/icu_ci.yml [hunk 1]
```diff
@@ -440,3 +440,32 @@ jobs:
...
+          # Perform an in-source build of icu4c
+          cd icu4c/source
+          ./runConfigureICU Linux
+          make -j2;
+          # Reference the paths in the in new build
+          export PATH=$PATH:$PREFIX/bin
+          export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PREFIX/lib
+
+      - name: Build and run samples
+          cd samples
```
Comment (gnrunge) [inline .github/workflows/icu_ci.yml:463]: This step lacks 'run: |'.

File: .github/workflows/icu_ci.yml [hunk 2]
```diff
@@ -440,3 +440,32 @@ jobs:
...
+      - name: ICU4C configure and build
+        env:
+          PREFIX: icu4c/source
+        run: |
+          # Perform an in-source build of icu4c
+          cd icu4c/source
+          ./runConfigureICU Linux
+          make -j2;
+          # Reference the paths in the in new build
+          export PATH=$PATH:$PREFIX/bin
```
Comment (gnrunge) [inline .github/workflows/icu_ci.yml:459]: This and the following statement are intended for the next step ('Build and run samples'). As such they should be in the next step, perhaps in an 'env:' clause.

**Review by gnrunge — CHANGES_REQUESTED** [review]

**Comment by sven-oly** [comment]
Please take another look.

**Review by gnrunge — DISMISSED** [review]

**Review by gnrunge — APPROVED** [review]

**Comment by jefgen** [comment]
Thanks for doing this! :)

**Pull request merged**